CK3toEU4/Source/EU4World/EU4World.cpp

```diff
@@ -1672,27 +1672,39 @@ void EU4::World::africanQuestion()
 			continue;
 
 		// check owners of pass ends.
-		std::vector<int> endProvinces = pass.getEndA();
-		endProvinces.insert(endProvinces.end(), pass.getEndB().begin(), pass.getEndB().end());
-		std::map<std::string, std::vector<int>> ownerProvinces;
+		bool hasStart = false;
+		bool hasEnd = false;
 
-		for (const auto& provinceID: endProvinces)
+		for (const auto& provinceID: pass.getEndA())
 		{
 			if (!provinces.find(provinceID)->second->getOwner().empty())
-				ownerProvinces[provinces.find(provinceID)->second->getOwner()].emplace_back(provinceID);
+				hasStart = true;
```

Comment:
by dropping ownerProvinces[TAG] you're only checking if *someone* has hold of required province. That will always be true since all provinces in ck2 scope map to eu4 scope.

You need to check if a particular tag has a hold of both ends.

Reply:
Changed